test/basic.test.js

```diff
@@ -58,8 +57,179 @@ test('Subscribed clients receive updates', async function (t) {
   var publisher = await helper.startClient()
 
   await publisher.publish(msg.topic, msg.payload, msg)
-  await helper.delay(2000)
+  await helper.delay(500)
 
   await Promise.all(subscribers.map(c => c.end()))
   await publisher.end()
+}
+
+test('Subscribed clients receive updates - QoS 1', async function (t) {
+  t.plan(10, 'each client should receive a message')
+  t.tearDown(helper.closeBroker)
+
+  await helper.startBroker()
+
+  await testQos(t, 1)
+})
+
+test('Subscribed clients receive updates - QoS 2', async function (t) {
+  t.plan(10, 'each client should receive a message')
+  t.tearDown(helper.closeBroker)
+
+  await helper.startBroker()
+  await testQos(t, 2)
+})
```

Comment:
You should be able to include `test` inside `testQoS` and remove all the duplication

Reply:
I already did this, isn't it good?

Reply:
You have two almost identical tests, one on line 66 and one on line 76. I would like this duplication to be removed by moving all content of those into the testQoS function.